File: Module 4/C/MapCount.c
#include <stdio.h>
#include <conio.h>
#include <string.h>
int main()
{
	char k[12]="convergence";
	int count=0,ct[12];
	char ch;
	for (int i = 0; k[i]!='\0'; ++i)
		ct[i]=0;
	for (int i = 0; k[i]!='\0'; ++i)
	{
		count=0;
		for (int j = 0; k[j]!='\0'; ++j)
			if (k[i]==k[j])
				count++;
		if (ct[i]==0)
			ct[i]=count;
	}
	for (int i = 0; k[i]!='\0'; ++i)
	{
		printf("%c\t%d\n",k[i],ct[i]);
	}
}

File: Module 4/JAVA/MapCount.java
import java.util.HashMap;

public class MapCount {
    public static void main(String[] args) {
        String k="convergence";
        int count=0;
        char ch;
        HashMap<Character,Integer> hm=new HashMap<Character,Integer>();
        for(char i='a'; i<='z'; i++)
        {
            count = 0;
            for(int j=0; j<k.length(); j++)
            {
                ch=k.charAt(j);
                if(ch==i)
                    count++;
            }
            if(count!=0)
            {
                hm.put(i,count);
            }
        }
        System.out.println(hm);
        HashMap<Integer,String> hm2=new HashMap<Integer,String>();
        for (Character key: hm.keySet())
        {
            if (!hm2.containsKey(hm.get(key)))
                hm2.put(hm.get(key),key.toString());
            else
                hm2.put(hm.get(key),(hm2.get(hm.get(key)).concat(','+key.toString())));
        }
        System.out.println(hm2);
    }
}